lib/app/models/testing_instance.rb

```diff
@@ -97,16 +97,32 @@ def waiting_on_sequence
         sequence_results.first(result: 'wait')
       end
 
-      def all_passed?
-        required_sequences = Inferno::Sequence::SequenceBase.subclasses.reject(&:optional?)
+      def all_test_cases(test_set_id)
+        self.module.test_sets[test_set_id.to_sym].groups.each_with_object([]) { |g, o| o.push(*g.test_cases) }
```

Comment:
```suggestion
        self.module.test_sets[test_set_id.to_sym].groups.flat_map(&:test_cases)
```
I think the above should be equivalent.